### PR TITLE
Add tenancy package and implement tenancy checks on SSL API endpoints.

### LINKED ISSUE #1763: Add Tenancy checks to traffic_ops_golang SSL API endpoints.
The traffic_ops_golang SSL keys API endpoints need Tenancy checks.

### PR DESCRIPTION
**Pull request by jrushford**
This adds tenancy checks to the fetch and add traffic_ops_golang SSL API endpoints.  Fixes #1763

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/880/
Test PASSed.

File: traffic_ops/traffic_ops_golang/deliveryservices_keys.go [hunk 1]
```diff
@@ -34,10 +34,63 @@ import (
...
+	tenantInfoStmt, err := db.Preparex("SELECT xml_id,tenant_id FROM deliveryservice where xml_id = $1")
+	if err != nil {
+		log.Errorf("deliverservice sql query error, xmlId '%s': %v\n", xmlId, err)
+		return nil, err
+	}
+
+	err = tenantInfoStmt.Get(&ds, xmlId)
+	switch {
+	case err == sql.ErrNoRows:
+		ds = DeliveryServiceTenantInfo{ID: -1, XMLID: "", TenantID: -1}
```
Comment (dangogh) [inline traffic_ops/traffic_ops_golang/deliveryservices_keys.go]: does -1 have special meaning here?   should just use the zero value for ds,  so this assignment isn't needed.

File: traffic_ops/traffic_ops_golang/deliveryservices_keys.go [hunk 2]
```diff
@@ -34,10 +34,63 @@ import (
...
+	}
+	return true, nil, http.StatusOK
+}
+
+// returns tenant information for a deliveryservice
+func getDeliveryServiceTenantInfo(xmlId string, db *sqlx.DB) (*DeliveryServiceTenantInfo, error) {
+	ds := DeliveryServiceTenantInfo{}
+	tenantInfoStmt, err := db.Preparex("SELECT xml_id,tenant_id FROM deliveryservice where xml_id = $1")
+	if err != nil {
+		log.Errorf("deliverservice sql query error, xmlId '%s': %v\n", xmlId, err)
```
Comment (dangogh) [inline traffic_ops/traffic_ops_golang/deliveryservices_keys.go]: spelling -- `deliveryservice`

File: traffic_ops/traffic_ops_golang/deliveryservices_keys.go [hunk 3]
```diff
@@ -34,10 +34,63 @@ import (
...
+	ID       int    `db:"id"`
+	XMLID    string `db:"xml_id"`
+	TenantID int    `db:"tenant_id"`
+}
+
+// returns true if the user has tenant access on this deliveryservice
+func (dsInfo DeliveryServiceTenantInfo) IsTenantAuthorized(user auth.CurrentUser, db *sqlx.DB) (bool, error) {
+	return auth.IsResourceAuthorizedToUser(dsInfo.TenantID, user, db)
+}
+
```
Comment (dangogh) [inline traffic_ops/traffic_ops_golang/deliveryservices_keys.go]: tenancy checks for deliveryservices are needed outside of the `_keys` code.  We should discuss where this should go..

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/888/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/903/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/904/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/905/
Test PASSed.

**Pull request merged**